### ISSUE TITLE
Check certificate subject and altnames

### ISSUE THREAD

**Issue by felicienfrancois** [dyaa/node-ssl-checker#13]
Currently, there is no error raised if the certificate does not match the requested hostname.

**Comment by felicienfrancois** [comment]
Basically, you should rely on `response.socket.authorized` which already contains result of validation

**Comment by dyaa** [comment]
@felicienfrancois are we talking about this case `https://wrong.host.badssl.com/`

**Comment by felicienfrancois** [comment]
@dyaa Exactly.
Using `response.socket.authorized` may also cover other cases of "invalid certificates"

**Comment by dyaa** [comment]
So, You expect if the `response.socket.authorized` was `false` to return an error instead of returning the remaining date? @felicienfrancois

**Comment by dyaa** [comment]
in this case the expired domain will fail as well, Currently we return the data object with negative `days_remaining`.

**Comment by felicienfrancois** [comment]
I suggest to add a field to returned payload so it won't makes any braking changes.
Example :
```json
{
	"valid" : true, // whether certificate pass ssl validation or not
	"days_remaining" : 90,
	"valid_from" : "issue date",
	"valid_to" : "expiry date"
}
```

**Comment by dyaa** [comment]
Published a new version `ssl-checker@1.2.0` with these changes, Please Verify.

**Comment by felicienfrancois** [comment]
@dyaa now I get an `SyntaxError: Unexpected token import` at index.js:3
It seems that you change requires to import which is not supported nativelly on my node version (8.9.3).

**Comment by dyaa** [comment]
Done, Check `ssl-checker@1.2.1`

**Comment by felicienfrancois** [comment]
Works well now. Thank you !
